flow/src/main/java/flow/History.java

```diff
@@ -207,7 +207,12 @@ private Builder(Collection<Entry> history) {
     }
 
     public Builder clear() {
-      history.clear();
+      // Clear by popping everything (rather than just calling history.clear()) to
+      // fill up entryMemory. Otherwise we drop view state on the floor.
+      while (!isEmpty()) {
```

Comment:
Chatted offline and realized that @pyricau has no context for this change. The problematic call to clear() is here:

https://github.com/square/flow/blob/ray/shame-about-that-view-state/flow/src/main/java/flow/Flow.java#L240

Basically, builders spawned from an existing History make an implicit promise hold on to view-state metadata as screens are popped off of them and possibley popped back on. The clear method needs to keep the same promise that the pop method does, and its failure to do so breaks Flow#preserveEquivalentPrefix